src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -0,0 +1,170 @@
+     * A permission with this rank can be granted to anybody including the
+     * untrusted guests. For most servers there is no difference between
+     * {@link #RANK_GUEST} and {@link #RANK_USER}.
+     *
+     * <p>Examples can be chat, MotD receive.</p>
+     */
+    static final String RANK_GUEST = "guest";
+    /**
+     * A permission with this rank can be granted to anybody that should have
+     * default user access. For most servers there is no difference between
+     * {@link #RANK_GUEST} and {@link #RANK_USER}.
+     *
+     * <p>Examples can be claim city plot, access to games and public warps.</p>
+     */
+    static final String RANK_USER = "user";
+    /**
+     * A permission with this rank should only be granted to known server staff.
+     * The permission's impact on the server is quite low.
+     *
+     * <p>Examples can be give items, teleport other players, access to support
+     * commands, globally mute other players.</p>
+     */
+    static final String RANK_STAFF = "staff";
+    /**
+     * A permission with this rank should only be granted to fully trusted
+     * server administrators. The permission's impact on the server may be high.
+     *
+     * <p>Examples can be create/delete worlds, grant permissions, restart
+     * server.</p>
+     */
+    static final String RANK_ADMIN = "admin";
+    /**
+     * A permission with this rank should normally not granted to anybody. This
+     * usually applies to debug permissions.
+     */
+    static final String RANK_NOBODY = "nobody";
+
+    /**
+     * Gets the permission id this description belongs too. May include
+     * placeholders {@code "<SomeParameter>"} such as in
+     * {@code "pluginId.give.item.<ItemType>"}.
+     *
+     * @return The permission id
+     */
+    String getId();
+
+    /**
+     * Gets a short description of the linked permission. May include a link to
+     * a more detailed description on the plugin's web page.
+     *
+     * @return A short description of the linked permission
+     */
+    Text getDescription();
+
+    /**
+     * Gets the human readable <b>suggested</b> rank the players should have
+     * before they should have access to this permission.
+     *
+     * <p><b>Note:</b> This is a suggestion for server owners and does not have
+     * any impact on the results of the permission checks.</p>
+     *
+     * @return The suggested player rank the permission should be granted to
+     */
+    Optional<String> getSuggestedRank();
+
+    /**
+     * Gets the owning plugin the permission belongs to.
+     *
+     * @return The owning plugin the permission belongs to
+     */
+    PluginContainer getOwner();
+
+    /**
+     * A builder for permission descriptions.
+     */
+    interface Builder {
+
+        /**
+         * Sets the permission id for the description this builder creates. May
+         * include placeholders {@code "<SomeParameter>"} such as in
+         * {@code "pluginId.give.item.<ItemType>"}.
```

Comment:
My current intention was that the server owner can read it as well as some methods to provide some basic syntax plugins can rely on. 

Let my explain this a bit.
The entire information included in a list of permissions like
* `pluginId.give.item.DIRT`
* `pluginId.give.item.GRAVEL`
* `pluginId.give.item.STONE`
* `pluginId.give.item.GRASS`

is also included in the single permission template `pluginId.give.item.<ItemType>`.
This also better for typos since if you type `pluginId.give.item.GASS` it can still match it to the template. Although the ItemType is not available on that server right now.
I thought of something like this:
````
PermissionDescriptionNode node = ROOTNODE;
String[] parts = permissionId.split(".", LITTERAL);
boolean matches = true;
for (String part : parts) {
    if (node.hasChild(part)) {
        node = node.getChild(part);
    } else if (node.hasTemplateChildren()) {
        node = node.getTemplateChildren();
    } else {
        matches = false;
        break;
    }
}
if (matches) {
    if (node.isMultiNode()) {
         showAmbigousMessage(player, node.getChoices(current));
    } else {
        showMessage(player, node.getDescription(), node.getChoices(parent), node.getChoices(children));
    }
} else {
    showNotFound(player, node.getChoices(parent + current + children));
}
````

`/perm description <permissionId>`

NodeTree:
- `myPlugin`
  - `<Command>`
  - `give`
    - `<ItemType>`
      - `fullStack`
    - `<Enchantment>`

Example run on `myPlugin.give.<ItemType>`:
* `node = ROOT`
* `node = node.getChild("myPlugin") // "myPlugin"`
* `node = node.getChild("give") // "myPlugin.give"`
* `node = node.getChild("<ItemType>") // "myPlugin.give.<ItemType>"`
* `showMessage(player, [current node], [parent node], [child nodes])`
* Outputs:
  - `Allows the usage of the specified <ItemType> in the /give command.`
  - `For a full list of possible ItemTypes see LINK`
  - `Suggested for: Staff`
  - `Provided by: MyPlugin`
  - `Parent permission: /perm description myPlugin.give`
  - `Available sub permissions: /perm description myPlugin.give.<ItemType>.fullStack`

Example run on `myPlugin.give.GASS`:
* `node = ROOT`
* `node = node.getChild("myPlugin") // "myPlugin"`
* `node = node.getChild("give") // "myPlugin.give"`
* `node ("myPlugin.give.GASS") does not exist`
* `node = node.getTemplateChildren() // "myPlugin.give" , ["<ItemType>", "<Enchantment>"]`
* `showAmbigousMessage(player, [current nodes])`
* Outputs:
  - `The specified permission is ambigous. Please choose:`
  - `/perm description myPlugin.give.<ItemType>`
  - `/perm description myPlugin.give.<Enchantment>`

Example run on `myPlugin.give.GASS.*`:
* `node = ROOT`
* `node = node.getChild("myPlugin") // "myPlugin"`
* `node = node.getChild("give") // "myPlugin.give"`
* `node "myPlugin.give.GASS" does not exist`
* `node = node.getTemplateChildren() // "myPlugin.give" , ["<ItemType>", "<Enchantment>"]`
* `nodes ("myPlugin.give" , ["<ItemType>", "<Enchantment>"] , "*") do not exist`
* `matches = false`
* `showNotFound(player, [parent, current & sub nodes])`
* Outputs:
  - `The specified permission was not found. Please choose:`
  - `/perm description myPlugin.give` // Parent
  - `/perm description myPlugin.give.<ItemType>` // Current (multi) node
  - `/perm description myPlugin.give.<Enchantment>`
  - `/perm description myPlugin.give.<ItemType>.fullStack` // Sub nodes of this (multi) node